### PR TITLE
Change paths to case2 rundir and exeroot

### PR DESCRIPTION
**Pull request by billsacks**
Previously, the code assumed that rundir and exeroot were subdirectories
of CIME_OUTPUT_ROOT. This is the case for CESM machines, but not for all
ACME machines. The new version doesn't make this assumption.

Test suite: scripts_regression_tests.py on hobart
Test baseline: n/a
Test namelist changes: none
Test status: bit for bit

Fixes ESMCI/cime#2031

User interface changes?: none

Update gh-pages html (Y/N)?: N

Code review:

### CODE REVIEW

**Comment by billsacks** [comment]
I kicked off testing last night, but the tests hung (probably a system issue). I'm redoing tests now, but wanted to go ahead and open this for review. I'll check the tests a little later this morning and post results.

File: scripts/lib/CIME/tests/case_fake.py [hunk 1]
```diff
@@ -97,9 +98,11 @@ def create_clone(self, newcase, keepexe=False, mach_dir=None, project=None,
         os.makedirs(newcaseroot)
         clone = self.copy(newcasename = newcasename, newcaseroot = newcaseroot)
         if cime_output_root is not None:
-            self.set_value('CIME_OUTPUT_ROOT', cime_output_root)
+            clone.set_value('CIME_OUTPUT_ROOT', cime_output_root)
+        if exeroot is not None:
+            clone.set_value('EXEROOT', exeroot)
         if rundir is not None:
-            self.set_value('RUNDIR', rundir)
+            clone.set_value('RUNDIR', rundir)
```
Comment (billsacks) [inline scripts/lib/CIME/tests/case_fake.py]: Change from self.set_value to clone.set_value was a mostly unrelated bug fix that just affects unit tests (and not in any substantive way).

File: scripts/lib/CIME/SystemTests/system_tests_compare_two.py [hunk 2]
```diff
@@ -282,9 +282,13 @@ def _get_output_root2(self):
...
         """
-        # Since case 2 has the same name as case1 its CIME_OUTPUT_ROOT must also be different
+        # Since case2 has the same name as case1, its CIME_OUTPUT_ROOT
+        # must also be different, so that anything put in
+        # $CIME_OUTPUT_ROOT/$CASE/ is not accidentally shared between
+        # case1 and case2. (Currently nothing is placed here, but this
+        # helps prevent future problems.)
         output_root2 = os.path.join(self._case1.get_value("CIME_OUTPUT_ROOT"),
-                                    self._case1.get_value("CASE"), "case2")
+                                    self._case1.get_value("CASE"), "case2_output_root")
```
Comment (billsacks) [inline scripts/lib/CIME/SystemTests/system_tests_compare_two.py]: I debated what to do about CIME_OUTPUT_ROOT. Now that we set RUNDIR and EXEROOT separately, nothing is placed here. Yet this directory is still created by the scripts. So we get a directory like `/glade/scratch/sacks/TESTNAME/case2_output_root` with nothing in it. I found this a little confusing. On the other hand, I'm nervous about leaving case2's CIME_OUTPUT_ROOT at its default value, because that opens the door for things from case2 to clobber things from case1 (in case anything is put in `$CIME_OUTPUT_ROOT/$CASE/` - since that location will be the same for case1 and case2).

As you'll see below, the solution I came up with was adding a README file to this location, so that if someone stumbles upon this directory, they'll be pointed to where they likely really want to go. But I'm open to suggestions if you feel there's a better solution.

**Comment by billsacks** [comment]
I was having system issues with the testing, but now have run scripts_regression_tests on hobart and all tests pass.

File: scripts/lib/CIME/SystemTests/system_tests_compare_two.py (hunk 2 above)
Comment (jgfouca) [reply scripts/lib/CIME/SystemTests/system_tests_compare_two.py]: @billsacks , case2 will involve 3 different directions, it's RUNDIR, EXEROOT (blddir), and it's case dir. I would expect the case dir to be under the case2 output root. Is this not right?

Comment (billsacks) [reply scripts/lib/CIME/SystemTests/system_tests_compare_two.py]: @jgfouca No, that's not right. case2's case directory gets nested under case1's case directory:

```python
        caseroot2 = os.path.join(caseroot1, "case2", casename2)
```

which, in general, bears no relationship to either the case1 or case2 CIME_OUTPUT_ROOT.

It's confusing, I know.

Comment (jgfouca) [reply scripts/lib/CIME/SystemTests/system_tests_compare_two.py]: OK, thanks.
